packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -309,20 +309,79 @@ export class ProfilesUtils {
+                    ),
+                },
+            }
+        );
+        if (response === yesGloballyButton || response === yesWorkspaceButton) {
+            const scope = response === yesGloballyButton ? vscode.ConfigurationTarget.Global : vscode.ConfigurationTarget.Workspace;
+            await SettingsConfig.setDirectValue(globals.SETTINGS_SECURE_CREDENTIALS_ENABLED, false, scope);
+            await vscode.commands.executeCommand("workbench.action.reloadWindow");
+        } else {
+            throw new imperative.ImperativeError({
+                msg: localize(
+                    "ProfileUtils.promptAndDisableCredentialManagement.error",
+                    // eslint-disable-next-line max-len
+                    "Failed to load credential manager. This may be related to Zowe Explorer being unable to use the default credential manager in a browser based environment."
+                ),
+            });
+        }
+    }
+
     /**
      * Use the default credential manager in Zowe Explorer and setup before use
      * @returns Promise<imperative.ProfileInfo> the object of profileInfo using the default credential manager
      */
     public static async setupDefaultCredentialManager(): Promise<imperative.ProfileInfo> {
-        ZoweLogger.trace("ProfilesUtils.setupDefaultCredentialManager called.");
-        ZoweLogger.info(
-            localize("ProfilesUtils.setupDefaultCredentialManager.usingDefault", "No custom credential managers found, using the default instead.")
-        );
-        await ProfilesUtils.updateCredentialManagerSetting(globals.ZOWE_CLI_SCM);
-        return new imperative.ProfileInfo("zowe", {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-            credMgrOverride: imperative.ProfileCredentials.defaultCredMgrWithKeytar(ProfilesCache.requireKeyring),
-        });
+        try {
+            ZoweLogger.trace("ProfilesUtils.setupDefaultCredentialManager called.");
+            ZoweLogger.info(
+                localize(
+                    "ProfilesUtils.setupDefaultCredentialManager.usingDefault",
+                    "No custom credential managers found, using the default instead."
+                )
+            );
+            await ProfilesUtils.updateCredentialManagerSetting(globals.ZOWE_CLI_SCM);
+            const defaultCredentialManager = imperative.ProfileCredentials.defaultCredMgrWithKeytar(ProfilesCache.requireKeyring);
+            const profileInfo = new imperative.ProfileInfo("zowe", {
+                credMgrOverride: defaultCredentialManager,
+            });
+            // Trigger initialize() function of credential manager to throw an error early if failed to load
+            await profileInfo.readProfilesFromDisk();
+            return profileInfo;
+        } catch (err) {
+            if (err instanceof imperative.ProfInfoErr) {
+                if (err.message.includes("Failed to initialize secure credential manager")) {
```

Comment:
Thanks for making the error handling as specific as possible 🙂 
We could check the error code instead of a specific message string here:
`if (err.code === ProfInfoErr.LOAD_CRED_MGR_FAILED)`

See https://github.com/zowe/imperative/blob/8be0b55559fc271d66a7a95646319575320a7c42/packages/config/src/ProfileInfo.ts#L930-L940